buildkite/resource_pipeline_team.go

```diff
@@ -0,0 +1,250 @@
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create team pipeline",
+			fmt.Sprintf("Unable to create team pipeline: %s", err.Error()),
+		)
+		return
+	}
+
+	// Update state with values from API response/plan
+	state.Id = types.StringValue(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.Id)
+	state.Uuid = types.StringValue(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.Uuid)
+	state.PipelineId = types.StringValue(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.Pipeline.Id)
+	state.TeamId = types.StringValue(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.Team.Id)
+	state.AccessLevel = types.StringValue(string(apiResponse.TeamPipelineCreate.TeamPipelineEdge.Node.PipelineAccessLevel))
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (tp *pipelineTeamResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state pipelineTeamResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiResponse, err := getNode(
+		tp.client.genqlient,
+		state.Id.ValueString(),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read team pipeline",
+			fmt.Sprintf("Unable to read team pipeline: %s", err.Error()),
+		)
+	}
+
+	// Convert from Node to getNodeNodeTeamPipeline type
+	if teamPipelineNode, ok := apiResponse.GetNode().(*getNodeNodeTeamPipeline); ok {
+		if teamPipelineNode == nil {
+			resp.Diagnostics.AddError(
+				"Unable to get team pipeline",
+				"Error getting team pipeline: nil response",
+			)
+			return
+		}
+		updateTeamPipelineResourceState(&state, *teamPipelineNode)
+		resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	} else {
+		// Resource not found, remove from state
+		resp.Diagnostics.AddWarning("Team pipeline resource not found", "Removing team pipeline from state")
+		resp.State.RemoveResource(ctx)
+	}
+
+}
+
+func (tp *pipelineTeamResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (tp *pipelineTeamResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state pipelineTeamResourceModel
+	var accessLevel string
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("access_level"), &accessLevel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	found := findTeamInPipelineNode(&tp.client.genqlient, &state)
+	if !found {
+		_, err := createTeamPipeline(
+			tp.client.genqlient,
+			state.TeamId.ValueString(),
+			state.PipelineId.ValueString(),
+			PipelineAccessLevels(accessLevel),
+		)
+
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Unable to create team pipeline",
+				fmt.Sprintf("Unable to create team pipeline: %s", err.Error()),
+			)
+			return
+		}
```

Comment:
This is an interesting case. Whats the reason for finding it first? If it doesn't exist it should be the job of the update to create it, that will be handled by terraform itself which will instead call create